src/targets/objc/native.js

```diff
@@ -1,65 +1,84 @@
 'use strict';
 
 var util = require('util');
+var objcHelpers = require('./helpers');
 
 module.exports = function (options) {
   var opts = util._extend({
-    timeout: '10'
+    timeout: '10',
+    indent: true
   }, options);
 
   var code = [];
 
-  // Dependencies
   code.push('#import <Foundation/Foundation.h>');
+
+  if (this.source.cookies.length) {
+    var cookies = this.source.cookies.map(function (cookie) {
+      return encodeURIComponent(cookie.name) + '=' + encodeURIComponent(cookie.value);
+    });
+    this.source.headersObj['Cookie'] = cookies.join('; ');
+  }
+
+  // Build headers (including cookies)
+  if (Object.keys(this.source.headersObj).length) {
+    code.push(null);
+    code.push(objcHelpers.nsDictionaryBuilder('headers', this.source.headersObj, opts.indent));
+  }
+
+  var parameters;
+
+  if (this.source.postData.mimeType && (this.source.postData.text || this.source.postData.params)) {
+    switch (this.source.postData.mimeType) {
+      case 'application/x-www-form-urlencoded':
+        parameters = this.source.postData.params;
+
+        code.push(null);
```

Comment:
`this.source.postData.mimeType` is always set as per [`src/index.js`](https://github.com/ahmadnassri/httpsnippet/blob/master/src/index.js#L21), defaults to application/octet-stream.